File: AspNetCore.ApiGateway.Client/IApiGatewayClient.cs
using Microsoft.AspNetCore.JsonPatch;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace AspNetCore.ApiGateway.Client
{
    public interface IApiGatewayClient
    {
        Task<TResponse> GetAsync<TResponse>(ApiGatewayParameters parameters);
        Task<TResponse> PostAsync<TPayload, TResponse>(ApiGatewayParameters parameters, TPayload data);
        Task PutAsync<TPayload>(ApiGatewayParameters parameters, TPayload data);
        Task<TResponse> PutAsync<TPayload, TResponse>(ApiGatewayParameters parameters, TPayload data);
        Task PatchAsync<TPayload>(ApiGatewayParameters parameters, JsonPatchDocument<TPayload> data)
            where TPayload : class;
        Task<TResponse> PatchAsync<TPayload, TResponse>(ApiGatewayParameters parameters, JsonPatchDocument<TPayload> data)
            where TPayload : class;
        Task DeleteAsync(ApiGatewayParameters parameters);
        Task<TResponse> DeleteAsync<TResponse>(ApiGatewayParameters parameters);
        Task<IEnumerable<Orchestration>> GetOrchestrationAsync(ApiGatewayParameters parameters);
    }
}

File: ts-aspnetcore-apigateway-client/IApiGatewayClient.ts
import { ApiGatewayParameters } from "./ApiGatewayParameters";
import { JsonPatchOperation } from "./JsonPatch";
import { Orchestration } from "./Orchestration";

export interface IApiGatewayClient {
    GetAsync<TResponse>(parameters: ApiGatewayParameters): Promise<TResponse>;
    PostAsync<TPayload, TResponse>(parameters: ApiGatewayParameters, data: TPayload): Promise<TResponse | null>;
    PutAsync<TPayload, TResponse>(parameters: ApiGatewayParameters, data: TPayload): Promise<TResponse | null>;
    PatchAsync<TResponse>(parameters: ApiGatewayParameters, data: JsonPatchOperation[]): Promise<TResponse | null>;
    DeleteAsync<TResponse>(parameters: ApiGatewayParameters): Promise<TResponse | null>;
    GetOrchestrationAsync(parameters: ApiGatewayParameters): Promise<Orchestration[]>;
}